app/src/main/java/com/example/iou/map/activities/GoogleMapsActivity.java

```diff
@@ -0,0 +1,187 @@
+import com.google.android.gms.maps.model.MarkerOptions;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class GoogleMapsActivity extends FragmentActivity implements OnMapReadyCallback {
+
+    public String URL;
+    private GoogleMap mMap;
+    private ActivityGoogleMapsBinding binding;
+    private LocationManager locationManager;
+    private final static String KEY_LOCATION = "location";
+    private double longitude;
+    private double latitude;
+    private double restaurantLatitude;
+    private double restaurantLongitude;
+    private List<RestaurantItem> nearbyRestaurants;
+    Location mCurrentLocation;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        binding = ActivityGoogleMapsBinding.inflate(getLayoutInflater());
+        setContentView(binding.getRoot());
+
+        // Used to store the restaurants nearby the user
+        nearbyRestaurants = new ArrayList<>();
+
+        // Obtain the SupportMapFragment and get notified when the map is ready to be used.
+        SupportMapFragment mapFragment = (SupportMapFragment) getSupportFragmentManager()
+                .findFragmentById(R.id.map);
+        mapFragment.getMapAsync(this);
+
+        if (savedInstanceState != null && savedInstanceState.keySet().contains(KEY_LOCATION)) {
+            // Since KEY_LOCATION was found in the Bundle, we can be sure that mCurrentLocation is not null
+            mCurrentLocation = savedInstanceState.getParcelable(KEY_LOCATION);
+        }
+
+        // Helps get the current location of the user
+        locationManager = (LocationManager) getSystemService(LOCATION_SERVICE);
+        if (ContextCompat.checkSelfPermission(GoogleMapsActivity.this, Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED &&
+                ContextCompat.checkSelfPermission(GoogleMapsActivity.this, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
+            ActivityCompat.requestPermissions(GoogleMapsActivity.this, new String[]{Manifest.permission.ACCESS_COARSE_LOCATION, Manifest.permission.ACCESS_FINE_LOCATION}, 1);
+        }
+
+        // Registers for location updates using the GPS
+        locationManager.requestLocationUpdates(LocationManager.GPS_PROVIDER, 25, 3, new LocationListener() {
+            @Override
+            public void onLocationChanged(@NonNull Location location) {
+
+                // Retrieve the longitude and latitude of the user
+                longitude = location.getLongitude();
+                latitude = location.getLatitude();
+
+                // Add a pin at user's current location
+                LatLng userCoordinates = new LatLng(latitude, longitude);
+                mMap.addMarker(new MarkerOptions().position(userCoordinates).title("Your Current Coordinates"));
```

Comment:
You may need to add a check somewhere before this to see if `mMap` has been set yet